### PR TITLE
:fire: sequence based strided rolling + faster feature func looping + improved tsfresh integrations + doc updates

### PR DESCRIPTION
**Pull request by jonasvdd**

### CODE REVIEW

**Comment by jvdd** [comment]
This PR also handles (partly) #31

**Comment by jvdd** [comment]
Not there yet for #15 but is a huge step in the right direction! :smiley:

File: tsflex/features/segmenter/__init__.py [hunk 1]
```diff
@@ -0,0 +1,13 @@
+# -*- coding: utf-8 -*-
+"""Series segmentation submodule."""
+
+__author__ = "Jonas Van Der Donckt"
+
+from ... import __pdoc__
```
Comment (jvdd) [inline tsflex/features/segmenter/__init__.py:6]: Should we import this and do nothing with it? @jonasvdd

**Comment by jonasvdd** [comment]
This is a major change, work still to do:

- [ ] update docs
- [ ] update example notebooks (also many to many)
- [ ] rerun benchmarking
- [ ] shill even more! 🔥

File: tsflex/features/segmenter/strided_rolling.py [hunk 2]
```diff
@@ -0,0 +1,371 @@
...
+
+        # 3. Construct the index ranges and store the series containers
+        np_start_times, np_end_times = self._construct_start_end_times()
+        self.series_containers = self._construct_series_containers(
+            series_list, np_start_times, np_end_times
+        )
+
+        # 4. Check the sparsity assumption
+        if not self.approve_sparsity:
+            last_container = self.series_containers[-1]
```
Comment (jvdd) [inline tsflex/features/segmenter/strided_rolling.py:125]: Why only check for the last container (= container that contains the last series of the series list)?
I would loop over all the containers :thinking:

File: tsflex/features/feature.py [hunk 3]
```diff
@@ -32,17 +32,20 @@ class FeatureDescriptor(FrozenClass):
...
+          sequence-window size. The series **must have a sequence-index**
+          e.g. a range-index, float-index, or int-index, but **not a time-index**.
         * If the window's type is a `pd.Timedelta`, the window size represents
-            the window-time.
-        * If a `str`, it represents a window-time-string. \n
+            the window-time. The passed data **must have a time-index**.
+        * If a `str`, it represents a window-time-string. The **passed data must have
+          a time-index**. \n
             .. Note::
                 When no time-unit is present in the string, it represents the stride
```
Comment (jvdd) [inline tsflex/features/feature.py]: Would perhaps remove this behavior & throw an error?

File: tsflex/features/utils.py [hunk 4]
```diff
@@ -2,13 +2,50 @@
...
+            earliest_stop = min(earliest_stop, series.index[-1])
+        return latest_start, earliest_stop
+    elif bound_method == "outer":
+        earliest_start = series_list[0].index[0]
+        latest_stop = series_list[0].index[-1]
+        for series in series_list[1:]:
+            earliest_start = min(earliest_start, series.index[0])
+            latest_stop = max(latest_stop, series.index[-1])
+        return earliest_start, latest_stop
+    elif bound_method == "first":
```
Comment (jvdd) [inline tsflex/features/utils.py:43]: Does this make sense? In line 330 of `feature_collection.py` it is called on the values of a dict...

**Review by jvdd — CHANGES_REQUESTED** [review]
If remarks are sorted out, we can merge this imo.
Very nice work! :smiley:

File: tsflex/features/utils.py (hunk 4 above)
Comment (jonasvdd) [reply tsflex/features/utils.py:52]: What do you mean?

File: tsflex/features/segmenter/__init__.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (jonasvdd) [reply tsflex/features/segmenter/__init__.py:6]: will check if `pdoc` differ when this is not done

[the remark this answers is not in the document]
Comment (jonasvdd) [reply tsflex/features/segmenter/__init__.py:6]: indeed, isn't needed, good catch! 👍🏾

File: tsflex/features/segmenter/strided_rolling.py (hunk 2 above)
Comment (jonasvdd) [reply tsflex/features/segmenter/strided_rolling.py:129]: good catch!

File: tsflex/features/utils.py (hunk 4 above)
Comment (jvdd) [reply tsflex/features/utils.py:52]: Dict is not ordered, so calling values returns the values (i.e., series) in a 'random' order. Then it does not make sense to use bound_method = "first"

Comment (jonasvdd) [reply tsflex/features/utils.py:52]: Indeed, shouldn't be like that!

File: tsflex/features/segmenter/strided_rolling.py [hunk 5]
```diff
@@ -293,11 +296,14 @@ def __init__(
         super().__init__(data, window, stride, *args, **kwargs)
 
     def _construct_output_index(self, name: str) -> pd.Index:
-        window_idx_offset = self._get_window_offset(self.window)
+        window_offset = self._get_window_offset(self.window)
+        # bool which indicates whether the `end` lies on the a boundary
```
Comment (jvdd) [inline tsflex/features/segmenter/strided_rolling.py:300]:
```suggestion
        # bool which indicates whether the `end` lies on the boundary
```

File: tsflex/features/utils.py (hunk 4 above)
Comment (jvdd) [reply tsflex/features/utils.py]: Resolved in commit cf0928a

**Review by jvdd — APPROVED** [review]

**Pull request merged**